syntax = "proto3";
option cc_enable_arenas = true;

package Ydb.Issue;
option java_package = "com.yandex.ydb";

// IssueMessage is a transport format for yql/essentials/public/issue library
message IssueMessage {
    message Position {
        uint32 row = 1;
        uint32 column = 2;
        string file = 3;
    }

    Position position = 1;
    string message = 2;
    Position end_position = 3;
    uint32 issue_code = 4;
    // Severity values from yql/essentials/public/issue/protos/issue_severity.proto
    // FATAL = 0;
    // ERROR = 1;
    // WARNING = 2;
    // INFO = 3;
    uint32 severity = 5;
    repeated IssueMessage issues = 6;
}
